### PR TITLE
Switch from Volkswagen testing to BMW testing

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
@mihails-strasuns-sociomantic : Whie compiling, I get the following warning:
`./src/swarm/Const.d(112): Warning: toHash() must be declared as extern (D) size_t toHash() const nothrow @safe, not const ulong()`

I see two problems here:
- `d1to2fix` will forbid those tokens.
- Compilation passed even with `-w` being passed on the command line

### CODE REVIEW

File: src/swarm/Const.d [hunk 1]
```diff
@@ -109,7 +109,7 @@ public struct NodeItem
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/Const.d:109]: Is the commit message right (`Since HEAD is currently using v5.x.x`)?

Comment (mathias-lang-sociomantic) [reply src/swarm/Const.d:109]: No you're right. Fixed.

**Comment by mathias-lang-sociomantic** [comment]
Added 2 bonus commits since I'm at it

**Comment by mihails-strasuns-sociomantic** [comment]
> Warning: toHash() must be declared as extern (D) size_t toHash() const nothrow @safe, not const ulong()

This is going to be a problem - we have enough trouble with `const`/`immutable` forced on us by runtime, caring about `@safe` is definitely not desired. I would even consider.

Does warning appear on 2.071 or later version?

**Comment by gavin-norman-sociomantic** [comment]
>I would even consider.

What would you consider?

**Comment by mihails-strasuns-sociomantic** [comment]
> What would you consider?

Contacting upstream to find a way to address this situation for code bases that don't want `@safe` :)

**Comment by mathias-lang-sociomantic** [comment]
> Does warning appear on 2.071 or later version?

Appears with `dmd-transitional` :(

**Comment by mathias-lang-sociomantic** [comment]
Although [according to the code](https://github.com/dlang/dmd/blame/v2.071.0/src/todt.c#L1283-L1295) we can "safely" ignore this warning.

**Comment by mihails-strasuns-sociomantic** [comment]
Can you turn submodule upgrade commit into fixup for sociomantic-tsunami/swarm@60156cc ?

I would ignore warning for now but keep problem in mind :(

**Comment by mihails-strasuns-sociomantic** [comment]
And source fixup should probably go as fixup for https://github.com/sociomantic-tsunami/swarm/commit/3c933052abec1a9aec87a0f21c7d3fab7d2e913a

**Comment by mathias-lang-sociomantic** [comment]
So basically all fixup ? :P

**Comment by mihails-strasuns-sociomantic** [comment]
Well, it is one big fixup PR, isn't it? :)

**Comment by mathias-lang-sociomantic** [comment]
Like this ?

**Comment by mihails-strasuns-sociomantic** [comment]
Hm, you seem to have put it all into single commit now - is that intentional?

**Comment by mathias-lang-sociomantic** [comment]
Only the first 2, which fixes the submodule update commit. They don't make sense separated if you use them to fixup.

**Comment by mihails-strasuns-sociomantic** [comment]
Some changes look related to compiler upgrade to 2.071, which was a separate commit. I notice one import  adjustment for example. False match?

**Comment by mathias-lang-sociomantic** [comment]
Hum I forgot I am using v2.071 by default now. I moved the protection change to its own commit. For the `const` change I *think* they belong here anyway.

**Review by mihails-strasuns-sociomantic — APPROVED** [review]
Thanks!

**Comment by mihails-strasuns-sociomantic** [comment]
@gavin-norman-sociomantic FYI I will make another swarm v5.x.x rebase myself after tagging v4.0.0 alpha in ocean this week, will double-check all tests at that point.

**Comment by mihails-strasuns-sociomantic** [comment]
Updated swarm v5.x.x to use ocean `v4.0.0-alpha.0`, squashed fixup commits and force pushed again.

**Pull request merged**